src/Sources/Coalescence.jl

```diff
@@ -28,31 +29,27 @@ update_coal_ints!(::AnalyticalCoalStyle, pdists::Array{ParticleDistribution{FT}}
 Updates the collision-coalescence integrals.
 """
 function update_coal_ints!(
-    ::AnalyticalCoalStyle,
+    cs::AnalyticalCoalStyle,
     pdists::Array{<:AbstractParticleDistribution{FT}},
     coal_data::NamedTuple,
 ) where {FT <: Real}
 
-    NProgMoms = [nparams(pdist) for pdist in pdists]
-    Nmom = maximum(NProgMoms)
-
     update_moments!(pdists, coal_data.moments)
     update_finite_2d_integrals!(pdists, coal_data.dist_thresholds, coal_data.moments, coal_data.finite_2d_ints)
 
     coal_data.coal_ints .= 0
-    for m in 1:Nmom
-
-        get_coalescence_integral_moment_qrs!(AnalyticalCoalStyle(), m - 1, NProgMoms, coal_data)
-
+    for m in 1:maximum(coal_data.NProgMoms)
```

Comment:
I think NProgMoms should not live within coal_data, but rather within p. In later functions it is passed as its own argument.